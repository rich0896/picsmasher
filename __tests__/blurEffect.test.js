/* eslint-disable no-undef */
// __tests__/blurEffect.test.js

const { BlurEffect } = require('../effects.js'); // Adjust the path as needed
const { applyEffectAndGetImageData } = require('../helpers/testHelpers.js');

describe('BlurEffect', () => {
    let ctx, canvas;

    beforeEach(() => {
        canvas = document.createElement('canvas');
        canvas.width = 10;
        canvas.height = 10;
        ctx = canvas.getContext('2d');

        // Fill the canvas with a solid color (e.g., rgb(50, 100, 150))
        ctx.fillStyle = 'rgb(50, 100, 150)';
        ctx.fillRect(0, 0, canvas.width, canvas.height);
    });

    test('should apply blur correctly at intensity 5', () => {
        const effect = new BlurEffect({ intensity: 5 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });

    test('should handle zero intensity without changes', () => {
        const effect = new BlurEffect({ intensity: 0 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });
});
