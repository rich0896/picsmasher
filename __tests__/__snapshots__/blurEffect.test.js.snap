// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlurEffect should apply blur correctly at intensity 5 1`] = `ImageData(10, 10, [50, 100, 150, 255, 50, 100, 150, 255, 50, 100...])`;

exports[`BlurEffect should handle zero intensity without changes 1`] = `ImageData(10, 10, [50, 100, 150, 255, 50, 100, 150, 255, 50, 100...])`;
